src/transports/http.c

```diff
@@ -32,8 +32,9 @@ static const char *post_verb = "POST";
 
 #define OWNING_SUBTRANSPORT(s) ((http_subtransport *)(s)->parent.subtransport)
 
-#define PARSE_ERROR_GENERIC	-1
-#define PARSE_ERROR_REPLAY	-2
+#define PARSE_ERROR_GENERIC	GIT_ERROR
+#define PARSE_ERROR_AUTHENTICATION	GIT_EAUTH
+#define PARSE_ERROR_REPLAY	1
```

Comment:
I think that makes sense, but then let's not redefine `GIT_` constants, let's just use them directly and keep `PARSE_ERROR` constants in the > 0 range.